plugins/coverimage.koplugin/main.lua

```diff
@@ -158,16 +197,440 @@ function CoverImage:onReaderReady(doc_settings)
     self:createCoverImage(doc_settings)
 end
 
+---------------------------
+-- cache handling functions
+---------------------------
+
+function CoverImage:getCacheFile()
+    local path, document_name = util.splitFilePathName(self.ui.document.file) --luacheck: no unsused
+    -- use document_name here. Title may contain characters not allowed on every filesystem (esp. vfat on /sdcard)
```

Comment:
Mimic Cache and use the md5 of the filename, maybe?

Reply:
Or even the docsetting's `partial_md5_checksum` if you don't want to re-compute it?

Reply:
Good point. partial_md5_checksum is not good, because the same document can have different cover image settings like background, format, stretching...
But md5 is fine.